infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/events/BrokerService.java

```diff
@@ -98,18 +97,26 @@ private void handle(BrokerStatusChangedEvent event) {
       LOG.error("Broker event skipped due to illegal content: {}", event);
       return;
     }
-    eventService.publish(new BrokerEvent(event, parseTooling(encodedTooling)));
-  }
-
-  @Nullable
-  private List<ChePlugin> parseTooling(String toolingString) {
-    if (!isNullOrEmpty(toolingString)) {
+    List<ChePlugin> tooling = null;
+    if (isNullOrEmpty(event.getError()) && !isNullOrEmpty(encodedTooling)) {
```

Comment:
I would move `!isNullOrEmpty(encodedTooling)` check to `parseTooling` method to avoid publishing event with null tooling and error.

Reply:
I moved this check out of the parse method not to break the single responsibility principle. I'm refactoring code a bit and adding unit tests